packages/pieces/discord/src/lib/common/index.ts

```diff
@@ -1,58 +1,175 @@
-        const options: { options: { value: string, label: string }[] } = { options: [] };
+      const request = {
+        method: HttpMethod.GET,
+        url: 'https://discord.com/api/v9/users/@me/guilds',
+        headers: {
+          Authorization: 'Bot ' + auth,
+        },
+      };
 
-        if (res.body.length === 0) return {
-            disabled: true,
-            options: [],
-            placeholder: "No guilds found, please add the bot to a guild first"
+      const res = await httpClient.sendRequest<Guild[]>(request);
+      const options: { options: { value: string; label: string }[] } = {
+        options: [],
+      };
+
+      if (res.body.length === 0)
+        return {
+          disabled: true,
+          options: [],
+          placeholder: 'No guilds found, please add the bot to a guild first',
         };
-        
-        await Promise.all(res.body.map(async (guild) => {
-            const requestChannels = {
+
+      await Promise.all(
+        res.body.map(async (guild) => {
+          const requestChannels = {
             method: HttpMethod.GET,
-            url: "https://discord.com/api/v9/guilds/" + guild.id + "/channels",
+            url: 'https://discord.com/api/v9/guilds/' + guild.id + '/channels',
             headers: {
-                "Authorization": "Bot " + auth,
-            }
-            };
+              Authorization: 'Bot ' + auth,
+            },
+          };
 
-            const resChannels = await httpClient.sendRequest<Channel[]>(requestChannels);
-            resChannels.body.forEach((channel) => {
+          const resChannels = await httpClient.sendRequest<Channel[]>(
+            requestChannels
+          );
+          resChannels.body.forEach((channel) => {
             options.options.push({
-                value: channel.id,
-                label: channel.name
-            });
+              value: channel.id,
+              label: channel.name,
             });
-        }));
+          });
+        })
+      );
+
+      return options;
+    },
+  }),
+  roles: Property.Dropdown<string>({
+    displayName: 'Roles',
+    description: 'List of roles',
+    required: true,
+    refreshers: ['guild_id'],
+    options: async ({ auth, guild_id }) => {
+      if (!auth) {
+        return {
+          disabled: true,
+          options: [],
+          placeholder: 'Please connect your bot first',
+        };
+      }
+
+      if (!guild_id) {
+        return {
+          disabled: true,
+          options: [],
+          placeholder: 'Please select a guild first',
+        };
+      }
+
+      const request = {
+        method: HttpMethod.GET,
+        url: `https://discord.com/api/v9/guilds/${guild_id}/roles`,
+        headers: {
+          Authorization: 'Bot ' + auth,
+        },
+      };
 
-        return options;
+      const res = await httpClient.sendRequest<Guild[]>(request);
+
+      console.log('Roles discord response', res.body);
```

Comment:
Let's remove this, as this would clutter the application logs, maybe output in case error?

Reply:
Whoops! Removed console log. Unnecessary